Sources/EduIDExpansion/Flows/Activity/ActivitMainNavigationDelegate.swift

```diff
@@ -0,0 +1,5 @@
+import UIKit
+
+protocol ActivitMainNavigationDelegate: AnyObject {
```

Comment:
Typo in name: missing y. Also, a view controller or coordinator declares its delegate. It shouldn't be a standalone thing like this.

Reply:
What is meant by 'declares'?

Reply:
ah like MainCoordinatorActivityNavigationDelegate as a name?

Reply:
```swift
protocol MyViewControllerDelegate: AnyObject {
  func myViewController(controller: MyViewController, wantsToKnowTheAnswerTo question: String) -> Int
}

class MyViewController: UIViewController {
  weak var delegate: MyViewControllerDelegate?

  func something() {
    let int = delegate?.myViewController(controller: self, wantsToKnowTheAnswerTo: "life")
  }
}
```

MyViewController defines the delegate protocol that whomever wants to use should provide so that it can do its thing. So if MyCoordinator wants to use MyViewController it should implement the delegate.

```swift
class MyCoordinator: Coordinator {
  func start() {
    let viewController = MyViewController()
    viewController.delegate = self
    // present viewController
  }
}

extension MyCoordinator: MyViewControllerDelegate {
  func myViewController(controller: MyViewController, wantsToKnowTheAnswerTo question: String) -> Int {
    return 42
  }
}
```

Reply:
thanks for the lovely example, I am interested if the naming also holds when the method doesn't have a parameter like to dismiss a flow:

`func myViewController(controller: UIViewController, wantsToDismiss parameterHereIsMissing: Any)`

Reply:
Or would it then be possible to extend the name of the method like:

`func myViewControllerDismissControllerFlow(controller: UIViewController)`

Reply:
Yes, the last one is the normal way to name that.